sw/device/silicon_creator/rom_exts/rom_ext_epmp.h

```diff
@@ -0,0 +1,52 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_ROM_EXTS_ROM_EXT_EPMP_H_
+#define OPENTITAN_SW_DEVICE_SILICON_CREATOR_ROM_EXTS_ROM_EXT_EPMP_H_
+
+#include <stdint.h>
+
+#include "sw/device/silicon_creator/lib/epmp.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * ROM_EXT ROM enhanced Physical Memory Protection (ePMP) library.
+ *
+ * The ePMP configuration is managed in two parts:
+ *
+ *   1. The actual hardware configuration held in CSRs
+ *   2. The in-memory copy of register values in `epmp_state_t` that is used
+ *      to verify the CSRs
+ *
+ * Every time the hardware configuration is updated the in-memory copy
+ * must also be updated. The hardware configuration is usually interacted
+ * with directly using the CSR library or assembly whereas the in-memory
+ * copy of the state should normally be modified using configuration functions
+ * from the silicon creator ePMP library.
+ *
+ * This separation of concerns allows the hardware configuration to be
+ * updated efficiently as needed (including before the C runtime is
+ * initialized) with the in-memory copy of the state used to double check the
+ * configuration as required.
+ */
+
+/**
+ * Unlocks the provided BLO0 image region with read-execute permissions.
+ *
+ * The provided ePMP state is also updated to reflect the changes made to the
+ * hardware configuration.
+ *
+ * @param state The ePMP state to update.
+ * @param image Region for executable sections in ROM_EXT image.
+ */
+void rom_ext_epmp_unlock_bl0_rx(epmp_state_t *state, epmp_region_t image);
```

Comment:
nit: It would be great if we could be consistent with the name we use for the first owner boot stage both in the code and the comments. For code we could go with `owner_stage` unless you think it is too verbose. For comments I think "first owner stage" or "first owner boot stage" are fine? @moidx, @mundaym WDYT?

context: this was called `bl0` in a few places but @moidx pointed out that this may cause some confusion since owners don't have to use it as a BL0, i.e. it can be the last boot stage.